<?xml version="1.0" encoding="utf-8"?>
<ui version="4.0">
 <author>
    SPDX-FileCopyrightText: none
    SPDX-License-Identifier: GPL-3.0-or-later
  </author>
 <class>GradientGeneratorConfigWidget</class>
 <widget class="QWidget" name="GradientGeneratorConfigWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>364</width>
    <height>225</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>175</height>
   </size>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <property name="usesScrollButtons">
      <bool>false</bool>
     </property>
     <widget class="QWidget" name="tabGeneral">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_8" stretch="0,1">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>10</number>
       </property>
       <property name="topMargin">
        <number>10</number>
       </property>
       <property name="rightMargin">
        <number>10</number>
       </property>
       <property name="bottomMargin">
        <number>10</number>
       </property>
       <item>
        <layout class="QFormLayout" name="formLayout">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
         </property>
         <property name="horizontalSpacing">
          <number>5</number>
         </property>
         <property name="verticalSpacing">
          <number>5</number>
         </property>
         <item column="0" row="0">
          <widget class="QLabel" name="labeShape">
           <property name="text">
            <string>Shape:</string>
           </property>
          </widget>
         </item>
         <item column="1" row="0">
          <widget class="QComboBox" name="comboBoxShape"/>
         </item>
         <item column="0" row="1">
          <widget class="QLabel" name="labelRepeat">
           <property name="text">
            <string>Repeat:</string>
           </property>
          </widget>
         </item>
         <item column="1" row="1">
          <widget class="QComboBox" name="comboBoxRepeat"/>
         </item>
         <item column="0" row="2">
          <widget class="QLabel" name="labelAntiAliasThreshold">
           <property name="text">
            <string>Anti-alias threshold:</string>
           </property>
          </widget>
         </item>
         <item column="1" row="2">
          <widget class="KisDoubleSliderSpinBox" name="sliderAntiAliasThreshold" native="true">
           <property name="focusPolicy">
            <enum>Qt::WheelFocus</enum>
           </property>
          </widget>
         </item>
         <item column="0" row="3">
          <widget class="QLabel" name="labelReverse">
           <property name="text">
            <string>Reverse:</string>
           </property>
          </widget>
         </item>
         <item column="1" row="3">
          <widget class="QCheckBox" name="checkBoxReverse">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabPosition">
      <attribute name="title">
       <string>Position</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_9" stretch="0,1">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>10</number>
       </property>
       <property name="topMargin">
        <number>10</number>
       </property>
       <property name="rightMargin">
        <number>10</number>
       </property>
       <property name="bottomMargin">
        <number>10</number>
       </property>
       <item>
        <layout class="QFormLayout" name="formLayout_2">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
         </property>
         <property name="horizontalSpacing">
          <number>5</number>
         </property>
         <property name="verticalSpacing">
          <number>10</number>
         </property>
         <item column="0" row="0">
          <widget class="QLabel" name="labelStartPosition">
           <property name="text">
            <string comment="Start position of the gradient">Start:</string>
           </property>
          </widget>
         </item>
         <item column="1" row="0">
          <layout class="QGridLayout" columnstretch="0,0,1" name="gridLayout" rowstretch="0,0">
           <property name="spacing">
            <number>5</number>
           </property>
           <item column="2" row="0">
            <widget class="QComboBox" name="comboBoxStartPositionXUnits"/>
           </item>
           <item column="2" row="1">
            <widget class="QComboBox" name="comboBoxStartPositionYUnits"/>
           </item>
           <item column="1" row="0">
            <widget class="QDoubleSpinBox" name="spinBoxStartPositionX">
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="minimum">
              <double>-50000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>50000.000000000000000</double>
             </property>
             <property name="value">
              <double>0.000000000000000</double>
             </property>
            </widget>
           </item>
           <item column="1" row="1">
            <widget class="QDoubleSpinBox" name="spinBoxStartPositionY">
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="minimum">
              <double>-50000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>50000.000000000000000</double>
             </property>
             <property name="value">
              <double>50.000000000000000</double>
             </property>
            </widget>
           </item>
           <item column="0" row="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>X:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item column="0" row="1">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Y:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item column="0" row="1">
          <widget class="QLabel" name="labelEndPosition">
           <property name="text">
            <string comment="End position of the gradient">End:</string>
           </property>
          </widget>
         </item>
         <item column="1" row="1">
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="spacing">
            <number>5</number>
           </property>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,1">
             <property name="spacing">
              <number>0</number>
             </property>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <property name="spacing">
                <number>5</number>
               </property>
               <item>
                <widget class="QRadioButton" name="radioButtonEndPositionCartesianCoordinates">
                 <property name="text">
                  <string>Cartesian Coordinates</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                 <attribute name="buttonGroup">
                  <string notr="true">buttonGroupEndPositionCoordinates</string>
                 </attribute>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="radioButtonEndPositionPolarCoordinates">
                 <property name="text">
                  <string>Polar Coordinates</string>
                 </property>
                 <property name="checked">
                  <bool>false</bool>
                 </property>
                 <attribute name="buttonGroup">
                  <string notr="true">buttonGroupEndPositionCoordinates</string>
                 </attribute>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QStackedWidget" name="stackedWidgetEndPosition">
             <property name="currentIndex">
              <number>0</number>
             </property>
             <widget class="QWidget" name="stackedWidgetPageEndPositionCartesian">
              <layout class="QGridLayout" columnstretch="0,0,1,1" name="gridLayout_3">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <property name="spacing">
                <number>5</number>
               </property>
               <item column="1" row="0">
                <widget class="QDoubleSpinBox" name="spinBoxEndPositionX">
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                 <property name="minimum">
                  <double>-50000.000000000000000</double>
                 </property>
                 <property name="maximum">
                  <double>50000.000000000000000</double>
                 </property>
                 <property name="value">
                  <double>100.000000000000000</double>
                 </property>
                </widget>
               </item>
               <item column="2" row="0">
                <widget class="QComboBox" name="comboBoxEndPositionXUnits"/>
               </item>
               <item column="3" row="1">
                <widget class="QComboBox" name="comboBoxEndPositionYPositioning"/>
               </item>
               <item column="3" row="0">
                <widget class="QComboBox" name="comboBoxEndPositionXPositioning"/>
               </item>
               <item column="2" row="1">
                <widget class="QComboBox" name="comboBoxEndPositionYUnits"/>
               </item>
               <item column="1" row="1">
                <widget class="QDoubleSpinBox" name="spinBoxEndPositionY">
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                 <property name="minimum">
                  <double>-50000.000000000000000</double>
                 </property>
                 <property name="maximum">
                  <double>50000.000000000000000</double>
                 </property>
                 <property name="value">
                  <double>50.000000000000000</double>
                 </property>
                </widget>
               </item>
               <item column="0" row="0">
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>X:</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item column="0" row="1">
                <widget class="QLabel" name="label_4">
                 <property name="text">
                  <string>Y:</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="stackedWidgetPageEndPositionPolar">
              <layout class="QVBoxLayout" name="verticalLayout_3">
               <property name="spacing">
                <number>5</number>
               </property>
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="KisDoubleSliderSpinBox" name="sliderEndPositionAngle" native="true"/>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,0,1">
                 <property name="spacing">
                  <number>5</number>
                 </property>
                 <item>
                  <widget class="QLabel" name="label_5">
                   <property name="text">
                    <string>Distance:</string>
                   </property>
                   <property name="alignment">
                    <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QDoubleSpinBox" name="spinBoxEndPositionDistance">
                   <property name="alignment">
                    <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                   </property>
                   <property name="minimum">
                    <double>-50000.000000000000000</double>
                   </property>
                   <property name="maximum">
                    <double>50000.000000000000000</double>
                   </property>
                   <property name="value">
                    <double>100.000000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QComboBox" name="comboBoxEndPositionDistanceUnits"/>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabGradientColors">
      <attribute name="title">
       <string>Gradient Colors</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4" stretch="0">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QScrollArea" name="scrollAreaGradientEditor">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Ignored">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarAlwaysOff</enum>
         </property>
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="KisGenericGradientEditor" name="widgetGradientEditor">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>356</width>
            <height>188</height>
           </rect>
          </property>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KisDoubleSliderSpinBox</class>
   <extends>QWidget</extends>
   <header location="global">kis_slider_spin_box.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>KisGenericGradientEditor</class>
   <extends>QWidget</extends>
   <header location="global">KisGenericGradientEditor.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>comboBoxShape</tabstop>
  <tabstop>comboBoxRepeat</tabstop>
  <tabstop>sliderAntiAliasThreshold</tabstop>
  <tabstop>checkBoxReverse</tabstop>
  <tabstop>spinBoxStartPositionX</tabstop>
  <tabstop>comboBoxStartPositionXUnits</tabstop>
  <tabstop>spinBoxStartPositionY</tabstop>
  <tabstop>comboBoxStartPositionYUnits</tabstop>
  <tabstop>radioButtonEndPositionCartesianCoordinates</tabstop>
  <tabstop>radioButtonEndPositionPolarCoordinates</tabstop>
  <tabstop>spinBoxEndPositionX</tabstop>
  <tabstop>comboBoxEndPositionXUnits</tabstop>
  <tabstop>comboBoxEndPositionXPositioning</tabstop>
  <tabstop>spinBoxEndPositionY</tabstop>
  <tabstop>comboBoxEndPositionYUnits</tabstop>
  <tabstop>comboBoxEndPositionYPositioning</tabstop>
  <tabstop>spinBoxEndPositionDistance</tabstop>
  <tabstop>comboBoxEndPositionDistanceUnits</tabstop>
 </tabstops>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroupEndPositionCoordinates"/>
 </buttongroups>
</ui>